src/test/java/org/junitpioneer/testkit/assertion/PropertiesAssert.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.testkit.assertion;
+
+import java.lang.reflect.Field;
+import java.util.Properties;
+
+import org.assertj.core.api.AbstractAssert;
+import org.junit.platform.commons.support.HierarchyTraversalMode;
+import org.junit.platform.commons.support.ReflectionSupport;
+
+/**
+ * Allow comparisons of java.util.Properties with optional awareness of their structure,
+ * rather than just treating them as Maps.  Object values, which are marginally supported
+ * by Properties, is supported in assertions.
+ *
+ */
+public class PropertiesAssert extends AbstractAssert<PropertiesAssert, Properties> {
+
+	/** No direct construction */
+	private PropertiesAssert(Properties actual) {
+		super(actual, PropertiesAssert.class);
+	}
+
+	/**
+	 * Begin an assertion
+	 *
+	 * @param actual The value the assert is made WRT.
+	 * @return Assertion instance
+	 */
+	public static PropertiesAssert assertThat(Properties actual) {
+		return new PropertiesAssert(actual);
+	}
+
+	/**
+	 * Assert Properties has the same effective values as the passed instance, but not
+	 * the same nested default structure.
+	 * <p>
+	 * Properties are considered <em>effectively same</em> if they have the same property
+	 * names returned by {@code Properties.propertyNames()} and the same values returned by
+	 * {@code getProperty(name)}.  Properties may come from the properties instance itself,
+	 * or from a nested default instance, indiscrimiately.
+	 * <p>
+	 * Properties partially supports object values, but return null for {@code getProperty(name)}
+	 * when the value is a non-string.  This assertion follows the same rules:  Any non-String
+	 * value is considered null for comparison purposes.
+	 *
+	 * @param expected The actual is expected to be effectively the same as this Properties
+	 * @return Assertion instance
+	 */
+	public PropertiesAssert isEffectivelyTheSameAs(Properties expected) {
```

Comment:
In AssertJ, "same" means `==`. Therefore, name it, e.g., `isEffectivelyEqualTo`? (Same applies to other "same" methods.)

Reply:
Changed.